core/domain/suggestion_services.py

```diff
@@ -2585,22 +2792,76 @@ def update_translation_contribution_stats_at_review(
             suggestion.change.language_code, suggestion.author_id, topic_id
         ))
 
-    if translation_contribution_stat_model is None:
-        accepted_translations_count = 0
-        accepted_translation_word_count = 0
-        rejected_translations_count = 0
-        rejected_translation_word_count = 0
-        accepted_translations_without_reviewer_edits_count = 0
+    translation_submitter_total_stat_model = (
+        suggestion_models.TranslationSubmitterTotalContributionStatsModel.get(
+            suggestion.change.language_code, suggestion.author_id
+        ))
 
-        if suggestion_is_accepted:
-            accepted_translations_count += 1
-            accepted_translation_word_count += content_word_count
-        else:
-            rejected_translations_count += 1
-            rejected_translation_word_count += content_word_count
-        if suggestion_is_accepted and not suggestion.edited_by_reviewer:
-            accepted_translations_without_reviewer_edits_count += 1
+    accepted_translations_count = 0
+    accepted_translation_word_count = 0
+    rejected_translations_count = 0
+    rejected_translation_word_count = 0
+    accepted_translations_without_reviewer_edits_count = 0
+
+    if suggestion_is_accepted:
```

Comment:
Actually we need to do this is `translation_contribution_stat_model` is None or `translation_contribution_stat_model` is None so rather than doing it twice I got it out of the if condition.

Reply:
I hope you mean `translation_contribution_stat_model` is None or `translation_submitter_total_stat_model` is None cases

But still don't we repeat this here https://github.com/kshitij01042002/oppia/blob/9b206e29e401987d6d078a0f00fc4b44a0dbb4d2/core/domain/suggestion_services.py#L3385C5-L3385C5

Can we move to separate function for the case of `translation_contribution_stat_model` is None or `translation_submitter_total_stat_model` is None cases?

Reply:
1. Yeah you are right, we need to calculate even if one of the case is None.

2. The code you have linked updates the existing model but if the model doesn't exist we have to calculate here.

3. I didn't quite get what exactly you mean?